src/azul/terraform.py

```diff
@@ -676,6 +676,32 @@ def tf_config(self, app_name):
                 key.replace('.', '_'): value
                 for key, value in bucket_notifications.items()
             }
+            # To prevent a race condition by Terraform, we make the bucket
+            # notifications depend on the related aws_lambda_permission and
+            # aws_lambda_function resources.
+            # https://stackoverflow.com/questions/67010382/terraform-error-putting-s3-notification-configuration-invalidargument-unable
```

Comment:
Not sure what purpose this link serves, since the solution used in this PR doesn't seem to be based on anything in either the question or any of the answers.

Reply:
If I'm missing something, and the PR is based on one of the answers, link to the answer instead of the question.

Reply:
> If I'm missing something, and the PR is based on one of the answers, link to the answer instead of the question.

The PR is based on the answer:

> So typically you want the S3 Notification to be the last thing that's deployed. Try making the S3 Notification depend on the Lambda too so that you're sure the Lambda gets deployed before the S3 Notification.

...however the comment "To prevent a race condition …" explains this without the need for a link, so i will remove the link and keep the comment.